### PR TITLE
test: pass PYTEST_ADDOPTS to pytest

### PR DESCRIPTION
**Pull request by tk0miya**
experimental PR for #4503.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -15,15 +15,15 @@ matrix:
     - python: '2.7'
       env:
         - TOXENV=du13
-        - PYTEST_ADDOPTS = --cov sphinx --cov-append --cov-config setup.cfg
+        - PYTEST_ADDOPTS="--cov ./ --cov-append --cov-config setup.cfg"
```
Comment (tk0miya) [inline .travis.yml:18]: Unfortunately, this is not recognized as envvar by travis. So it was not defined on runtime.

File: .travis.yml [hunk 2]
```diff
@@ -15,15 +15,15 @@ matrix:
...
     - python: '3.4'
       env: TOXENV=py34
     - python: '3.5'
       env: TOXENV=py35
     - python: '3.6'
       env:
         - TOXENV=py36
-        - PYTEST_ADDOPTS = --cov sphinx --cov-append --cov-config setup.cfg
+        - PYTEST_ADDOPTS="--cov ./ --cov-append --cov-config setup.cfg"
```
Comment (tk0miya) [inline .travis.yml:26]: The example of codecov says `./` is better.
https://github.com/codecov/example-python#pytest

File: tox.ini [hunk 3]
```diff
@@ -5,7 +5,7 @@ envlist = docs,flake8,mypy,coverage,py{27,34,35,36,py},du{11,12,13,14}
 [testenv]
 usedevelop = True
 passenv =
-    https_proxy http_proxy no_proxy PERL PERL5LIB
+    https_proxy http_proxy no_proxy PERL PERL5LIB PYTEST_ADDOPTS
```
Comment (tk0miya) [inline tox.ini:8]: `PYTEST_ADDOPTS` were not passed to pytest command. So this allows it.

File: .travis.yml [hunk 4]
```diff
@@ -46,4 +46,4 @@ script:
   - tox -- -v
 
 after_success:
-  - codecov
+  - if [[ -e .coverage ]]; then codecov -e $TOXENV; fi
```
Comment (tk0miya) [inline .travis.yml:49]: Send coverages only if major coverage, and tag it as `$TOXENV`.

**Comment by stephenfin** [comment]
@tk0miya Nice work! I was at FOSDEM last weekend so have been busy 😞

**Pull request merged**